### PR TITLE
conftest: add uses_pexpect mark

### PR DESCRIPTION
**Pull request by blueyed**
Pulled out of https://github.com/pytest-dev/pytest/pull/5313 - is this something we want to have?

### CODE REVIEW

File: testing/conftest.py [hunk 1]
```diff
@@ -23,7 +24,12 @@ def pytest_collection_modifyitems(config, items):
             # (https://github.com/pytest-dev/pytest/issues/5070)
             neutral_items.append(item)
         else:
-            if any(x for x in fixtures if x in slow_fixturenames):
+            if "testdir" in fixtures and spawn_names.intersection(
+                item.function.__code__.co_names
+            ):
+                slow_items.append(item)
+                item.add_marker(pytest.mark.uses_pexpect)
+            elif any(x for x in fixtures if x in slow_fixturenames):
```
Comment (blueyed) [inline testing/conftest.py]: Could be simplified given that `testdir` is the only slow fixture.

**Comment by blueyed** [comment]
I think a generic feature like `-k using_fixture(foo)`, or `--using-fixture foo` would be better.

File: tox.ini [hunk 2]
```diff
@@ -175,6 +175,7 @@ markers =
     baz
     # conftest.py reorders tests moving slow ones to the end of the list
     slow
+    uses_pexpect
```
Comment (nicoddemus) [inline tox.ini]: I would add a quick description here 👍

**Comment by nicoddemus** [comment]
> --using-fixture foo would be better.

Yep, I've seen the need for this myself... should we open a new issue for it?

**Comment by blueyed** [comment]
> Yep, I've seen the need for this myself... should we open a new issue for it?

Yeah.

File: tox.ini (hunk 2 above)
Comment (blueyed) [reply tox.ini]: Well, it is rather self-explaining, isn't it?

**Comment by blueyed** [comment]
Marking WIP to not merge it yet - there is no use for it currently really, given that we're likely to change the pexpect factor for CI (#5313, it would make sense to use `-m uses_pexpect` if jobs are being kept separate).

File: tox.ini (hunk 2 above)
Comment (blueyed) [reply tox.ini]: Ok, done - also added a new "slowest_items" category.

**Pull request merged**